uPortal-webapp/src/main/webapp/scripts/search/autosuggestHandler.js

```diff
@@ -97,36 +94,63 @@
                 });
             },
             /* Processor to handle /api/portletList JSON results */
-            portletListProcessor: function(portletListJson, urlPattern) {
-                return portletListProcessorFunc(
+            portletListProcessor: function (portletListJson, urlPattern) {
+                return portletListProcessorFunction(
                     portletListJson.registry,
                     urlPattern
                 );
             },
         };
     };
 
+    /**
+     * Simple function to escape specific HTML characters so they aren't
+     * a problem. &, quote, and single quote are handled by jQuery so
+     * only need to deal with < and >
+     * @param {string} str - unescaped input
+     * @return {string} escaped output
+     */
+    function htmlEscape(string_) {
+        return String(string_).replaceAll('<', '&lt;').replaceAll('>', '&gt;');
+    }
+
+    /**
+     * Helper method for making it a little easier to format the output in the menu
+     * @param  {object} item A JavaScript Object containing the item values
+     * @return {string}      Returns a formatted string that will be injected into the menu
+     */
+    var formatOutput = function (item) {
+        return (
+            '<a><span class="autocomplete-header">' +
+            htmlEscape(item.title) +
+            '</span><br>' +
+            htmlEscape(item.desc) +
+            '</a>'
+        );
+    };
```

Comment:
these are pure functions, and were extracted up a level to avoid unnecessary reallocation.